go-controller/pkg/ovn/egressgw.go

```diff
@@ -368,13 +369,30 @@ func (oc *Controller) addPerPodGRSNAT(pod *kapi.Pod, podIfAddrs []*net.IPNet) er
 			// If the type is snat and the logical IP is the same, but external IP is different,
 			// even with --may-exist, the add may error out. this is because, for snat,
 			// (type, router, logical ip) is considered a key for uniqueness
-			stdout, stderr, err := util.RunOVNNbctl("--if-exists", "lr-nat-del", gr, "snat", podIP+mask,
-				"--", "lr-nat-add",
-				gr, "snat", gwIP, podIP+mask)
+			// Combining lr-nat-del and lr-nat-add will fail OVSDB transaction check because
+			// the entry will already exist so need to do these as two separate transactions
+			// Therefore we need to iterate through existing SNAT rules on GR and decide if we need to delete/add
+			// TODO(trozet) use go-bindings
+			stdout, stderr, err := util.RunOVNNbctl("lr-nat-list", gr)
 			if err != nil {
-				return fmt.Errorf("failed to create SNAT rule for pod on gateway router %s, "+
+				return fmt.Errorf("failed to get SNAT rules on gateway router %s, "+
 					"stdout: %q, stderr: %q, error: %v", gr, stdout, stderr, err)
 			}
+			if len(stdout) > 0 {
+				reg := regexp.MustCompile(fmt.Sprintf(`snat\s+%s\s+%s`, gwIP, podIP))
+				if !reg.MatchString(stdout) {
```

Comment:
nit: since you are not reusing the regex, you can use directly `regexp.Match`

Reply:
will do

Reply:
going to leave this for now as regexpMatch returns bool and error, so doesnt seem any more convenient